### PR TITLE
Disallow whitespace in parameter names in design2params

### PR DESCRIPTION
**Pull request by berland**
Solves #243

### CODE REVIEW

File: semeio/jobs/design2params/design2params.py [hunk 1]
```diff
@@ -203,6 +203,14 @@ def _validate_design_matrix(design_matrix):
     if len(empties) > 0:
         raise SystemExit("Design matrix contains empty cells {}".format(empties))
 
+    # Look for initial or trailing whitespace in column headers. This
+    # is disallowed as it can create user confusion and has no use-case.
+    for col_header in design_matrix:
+        if col_header != col_header.strip():
+            raise SystemExit(
+                'Column header "{}" contains whitespace.'.format(col_header)
```
Comment (markusdregi) [inline semeio/jobs/design2params/design2params.py:211]: Perhaps we should change the message to say that is has initial or trailing whitespaces? Would be good if the condition and the message corresponds ;)

File: tests/jobs/design2params/test_design2params.py [hunk 2]
```diff
@@ -424,3 +424,67 @@ def test_pair_cell_values(cellvalues, expected_parameters_strs, tmpdir):
...
+
+
+@pytest.mark.parametrize(
+    "paramnames",
+    (
+        [
+            ["REAL", "SOMEKEY "],
+            ["REAL", " SOMEKEY"],
+            ["REAL", " SOMEKEY "],
+            ["REAL", "\tSOMEKEY"],
```
Comment (markusdregi) [inline tests/jobs/design2params/test_design2params.py:436]: Tabs :heart:

File: semeio/jobs/design2params/design2params.py (hunk 1 above)
Comment (berland) [reply semeio/jobs/design2params/design2params.py:211]: Yes. Good point!

File: tests/jobs/design2params/test_design2params.py (hunk 2 above)
Comment (berland) [reply tests/jobs/design2params/test_design2params.py:436]: Not sure it is possible to even inject through Excel..

**Review by markusdregi — APPROVED** [review]
Besides fixing the message this looks good :+1:

**Comment by berland** [comment]
Ready to 🚢

**Pull request merged**